File: April/22th_April/Smaller Sum.cpp
class Solution{
public:
    vector<long long> smallerSum(int n,vector<int> &arr){
        // Code here
         vector<long long> sum(n); vector<int> s;
        for(int x: arr) s.push_back(x);
        sort(begin(s),end(s));
        for(int i = 0; i < n; i++) 
            sum[i] +=  (i>0) ? sum[i-1] + s[i] : s[i];
            
        vector<long long> ans;
        for(int x: arr) {
           int l = 0, r = n-1, idx = -1;
           while(l <= r) {
               int mid = l + (r-l)/2;
               if(x > s[mid]) idx = mid , l = mid+1;
               else r = mid - 1;
           }
           ans.push_back(idx >= 0 ? sum[idx]: 0);
        }
        return ans;
    }
};

File: April/22th_April/Smaller Sum.java
class Solution
{
    public long[] smallerSum(int n,int arr[])
    {
        
               long[] pref=new long[n];
        long[] ans=new long[n];
        int[] asc=new int[n];
        for (int i = 0; i < arr.length; i++) {
            asc[i]=arr[i];
        }
        Arrays.sort(asc);
        pref[0]=asc[0];
        for (int i = 1; i < n; i++) {
           pref[i]= pref[i-1] + asc[i];
        }
        for (int i = 0; i < asc.length; i++) {
            if (asc[0]==arr[i]) {
                ans[i]=0;
                continue;                
            }
            int l=0,r=n-1;
            while (l<r) {
                int mid=(l+r+1)/2;
                if(asc[mid]>=arr[i]){
                    r=mid-1;
                }
                else{
                    l=mid;
                }
            }
            ans[i]=pref[l];
        }

        return ans;
    }
}